### DISCUSSION TITLE
adding support for BsonDiscriminator

### DISCUSSION THREAD

**Discussion by chlegou** [flamingock/mongock#32] [Ideas]
In polymorphism with abstract class, sometimes we want to map our entity to subclass based on property value, not by "_class" default mapping strategy. For this, we have `@BsonDiscriminator(key = "type", value = "USER")`

Please look for this link for more details:  
https://zx77.medium.com/mongodb-java-driver-for-polymorphism-8d8a9e28ec24

When I do so,:  
- it always map relating to "_class" property not "type" property.
-  after removing "_class" (trying to force mapping of property), it tries creating an instance from an abstract class and throws error.

```
     // BaseUser is abstract class
     org.springframework.beans.BeanInstantiationException:: Failed to instantiate BaseUser using constructor public BaseUser() with arguments. 
```

Please add support to force mapping strategy following  `@BsonDiscriminator` prior to "_class".

Thanks.

**Comment by dieppa** [comment]
Hello @chlegou , we really appreciate your suggestion.We are currently busy on other important priorities and won't be able to address this in the near future. We suggest you to try a PR and I'll personally review and help you in anything you need.

Thanks
